diff --git a/src/ssli_mbedtls.c b/src/ssli_mbedtls.c
--- a/src/ssli_mbedtls.c
+++ b/src/ssli_mbedtls.c
@@ -54,16 +54,18 @@
 
 const int ciphers[] =
 {
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3)
 	MBEDTLS_TLS1_3_AES_256_GCM_SHA384,
 	MBEDTLS_TLS1_3_CHACHA20_POLY1305_SHA256,
 	MBEDTLS_TLS1_3_AES_128_GCM_SHA256,
+#endif
 	MBEDTLS_TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
 	MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
 	MBEDTLS_TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
 	MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
 	MBEDTLS_TLS_RSA_WITH_AES_256_CBC_SHA,
 	MBEDTLS_TLS_RSA_WITH_AES_128_CBC_SHA,
-    0
+	0
 };
 
 #if !defined(MBEDTLS_USE_PSA_CRYPTO)
@@ -437,7 +439,9 @@
 		} else if (rc == MBEDTLS_ERR_X509_CERT_VERIFY_FAILED) { /* Allow this (selfsigned etc) */
 			return 0;
 		} else {
-			Log_warn("SSL handshake failed: %d", rc);
+			char err[128];
+			mbedtls_strerror(rc, err, sizeof(err));
+			Log_warn("SSL handshake failed: %s", err);
 			return -1;
 		}
 	}
